### PR TITLE
Docs cleanup

### PR DESCRIPTION
**Pull request by wtrocki**
## Motivation

Reorganize documentation to support new architecture.
This is wip but sharing for early feedback and to be able to link people to the unmerged docs

### CODE REVIEW

File: docs/plugins/creating-plugin.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+id: codegenplugin
+title: Creating your own plugin
+---
+
+Graphback allows to create your own plugins that can affect generated schema 
+and create files in filesystem.
+
+Each plugin needs to be an new npm package that will implement GraphbackPlugin.
+
+## Building your plugin
```
Comment (wtrocki) [inline docs/plugins/creating-plugin.md]: @cecobask FYI

**Comment by wtrocki** [comment]
Current layout:


Screenshot 2020-02-13 at 17 15 07
Screenshot 2020-02-13 at 17 15 19

**Comment by wtrocki** [comment]
@craicoverflow I will look for very early feedback on layout that was proposed and also on the getting started which I will refine first. 

I'm not happy with getting started experience.

File: docs/commands.md [hunk 2]
```diff
@@ -14,16 +14,20 @@ Execute `graphback` in your shell for more information
...
+> **Note** When creating configuration please review all paths as they might not reflect your project stucture
 
 ### Generate
 
   ```bash
   graphback generate
   ```
-  Generate command will generate schema and resolvers for your GraphQL server based on your data model.
+
+  Generate command will execute generation process that will gener GraphQL server based on your data model.
```
Comment (craicoverflow) [inline docs/commands.md]:
```suggestion
  Generate command will execute generation process that will generate GraphQL server based on your data model.
```

**Comment by wtrocki** [comment]
Website published. Docs available here: https://graphback.dev/docs/next/gettingstarted

Leaving that for review

**Comment by wtrocki** [comment]
@darahayes @kingsleyzissou @pwright 

Pinging you here to get general feedback on the layout of the new graphback docs we have build: graphback.dev/docs/next/gettingstarted

Let me know if you do not feel lost and can get your head around what graphback is just from scooping from the titles of the sidebar. 
Docs are still in very rough state and missing a lots of content but we have build basic layout that we want to keep.

**Comment by craicoverflow** [comment]
AWS Amplify Getting Started is concise and to the point: https://aws-amplify.github.io/docs/

There are three or four lines of text before you are shown how to install the CLI.

Our first [Getting Started page](https://graphback.dev/docs/next/gettingstarted) has too much text.

I as a developer will want to install the CLI as soon as I can. I don't want to navigate the system for too long to simply find how I can install it.

I would suggest have a repeat of `Create App in 2 minutes` guide on the homepage too, but that is unversioned so I don't think it would suffice.

**Comment by wtrocki** [comment]
There is a lot in your comment so going to split that to sections:

1) Getting started is too long

I will split that into 2 chapters (second one starts with `Graphback Code generation

2) Point you have made is mostly related to our website.
We need to improve our website (both offix and graphback)

3) Getting started docs is just the first landing page.
Maybe we should be landing on the 2 minutes docs instead?
Changing the structure of the intro?

4) Asking people to install graphql-cli when graphback cli is active is missleading. We need to have fully functional CLI itself and port stuff to graphql-cli. Restoring templates idea?

As amplify was used here to provide context: 
Amplify is DataSync. Graphback will be just tool under the hood so it will not have IOS, Android etc. guides for the moment. Document that was linked is part o the webpage. Actual docs has some text: https://aws-amplify.github.io/docs/js/start?ref=amplify-js-btn&platform=purejs

**Comment by craicoverflow** [comment]
>     2. Getting started docs is just the first landing page.
>        Maybe we should be landing on the 2 minutes docs instead?

I think this would be a good action to take. 2 minute docs is the fastest way to getting started.

Let's hear feedback from others before making this change though.

**Comment by joaedwar** [comment]
The structure looks good, but I agree with @craicoverflow, there is a lot of writing (a bit verbose) and I think it is a bit confusing at times or does not add a ton of value to the user's experience.

For instance, this sentence should be rewritten:  After project is created please follow project README file that might differ depending on the template that was chosen on startup.
 
Some other things at first glance: 
- 'Adding Graphback to existing project using Graphback CLI section', I think can be re-ordered. Have intro/ context then commands

- 'Creating and updating your database from your data model' can be broken up into easier to digest segments.

**Comment by pwright** [comment]
Took a look and got stuck at  ```graphql codegen``` step, 
one comment I'd make is to promote 'What Graphback can do for you?' to top of home page.

**Comment by wtrocki** [comment]
@pwright We will be moving away from the graphql-cli towards our own in our docs since it takes us some time to integrate that back

**Comment by wtrocki** [comment]
Thank you all for such amazing feedback.
CircleCI is broken now. Merging to incorporate other changes easiy.

**Comment by wtrocki** [comment]
@joaedwar Thank you so much. Going to apply all in separate PR as this was closed

**Comment by wtrocki** [comment]
I have applied changes all in separate PR. Thank you so much for such amazing fixes.

**Pull request merged**